src/deluge/io/midi/midi_device_manager.cpp

```diff
@@ -565,11 +565,18 @@ void ConnectedUSBMIDIDevice::bufferMessage(uint32_t fullMessage) {
 }
 
 bool ConnectedUSBMIDIDevice::hasBufferedSendData() {
-	// must me the same unsigned type as ringBufWriteIdx/ringBufReadIdx
+	// must be the same unsigned type as ringBufWriteIdx/ringBufReadIdx
 	uint32_t queued = ringBufWriteIdx - ringBufReadIdx;
 	return queued > 0;
 }
 
+int ConnectedUSBMIDIDevice::sendBufferSpace() {
+	// must be the same unsigned type as ringBufWriteIdx/ringBufReadIdx
+	uint32_t queued = ringBufWriteIdx - ringBufReadIdx;
+	// each 4-byte MIDI-USB message contains 3 bytes of serial MIDI data
+	return (MIDI_SEND_BUFFER_LEN_RING - queued) * 3;
```

Comment:
Won't this overreport the buffer to be up to 3 times the allocated size? If used to pack the buffer it could lead to run over the array bounds.

Reply:
no, for the reason explained in the comment just above.

Reply:
Ah, I overlooked the buffer being 32 bit values and this function reporting bytes available not "slots". lgtm